go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -242,6 +251,48 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 		return errs.Error()
 	}
 
+	// Look for any other tablets claiming to be master and fix them up on a best-effort basis
+	tabletMap, err := agent.TopoServer.GetTabletMapForShard(ctx, masterTablet.Keyspace, masterTablet.Shard)
+	if err != nil {
+		log.Errorf("ignoring error %v from GetTabletMapForShard so that we can process any partial results", err)
+	}
+
+	for _, tabletInfo := range tabletMap {
+		alias := tabletInfo.Tablet.Alias
+		if !topoproto.TabletAliasEqual(alias, agent.TabletAlias) && !topoproto.TabletAliasEqual(alias, oldMasterAlias) && tabletInfo.Tablet.Type == topodatapb.TabletType_MASTER {
+			log.Infof("finalizeTabletExternallyReparented: updating tablet record for another old master: %v", alias)
+			wg.Add(1)
+			go func(alias *topodatapb.TabletAlias) {
+				defer wg.Done()
+				var err error
+				tab, err := agent.TopoServer.UpdateTabletFields(ctx, alias,
+					func(tablet *topodatapb.Tablet) error {
+						if tablet.Type == topodatapb.TabletType_MASTER {
+							tablet.Type = topodatapb.TabletType_REPLICA
+							return nil
+						}
+						return topo.NewError(topo.NoUpdateNeeded, alias.String())
+					})
+				if err != nil {
+					errs.RecordError(err)
+					return
+				}
+				// tab will be nil if no update was needed
+				if tab != nil {
+					log.Infof("finalizeTabletExternallyReparented: Refresh state for tablet: %v", topoproto.TabletAliasString(tab.Alias))
+					tmc := tmclient.NewTabletManagerClient()
```

Comment:
We can share the TMC among all goroutines. Also we should probably `defer tmc.Close()`.

Reply:
this requires me to add an import to a bunch of test files, but I think it's still worth it.